src/annotator/util/shadow-root.js

```diff
@@ -43,5 +44,18 @@ export function createShadowRoot(container) {
     applyFocusVisible(shadowRoot);
   }
 
+  stopEventPropagation(shadowRoot);
```

Comment:
I wasn't sure if this was needed given that only events that are specifically marked as `composed` propagate across shadow DOM boundaries, but [according to MDN](https://developer.mozilla.org/en-US/docs/Web/API/Event/composed) this is the case for all browser-generated events:

> All UA-dispatched UI events are composed (click/touch/mouseover/copy/paste, etc.).

Reply:
You are absolutely correct.

In one of my tests, it took me a while to emulate the browser-generate event `element.click()` which propagated through the shadow DOM, with the `new Event('click')` which didn't. The reason was the `composed` property.

Reply:
I will add a comment in these line for future reference:
 https://github.com/hypothesis/client/blob/cb4af8934c37253059c0de71945eb0b3378ae8bb/src/annotator/util/test/shadow-root-test.js#L69